File: app/src/main/java/com/lis/player_java/ui/AuthorizationFragment.java
package com.lis.player_java.ui;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.lis.player_java.R;
import com.lis.player_java.data.network.Authorization;
import com.lis.player_java.databinding.FragmentAuthorizationBinding;

import java.util.Map;

public class AuthorizationFragment extends Fragment {

    private FragmentAuthorizationBinding binding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentAuthorizationBinding.inflate(inflater, container, false);
        Log.e("onCreate", "author");
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        bindElement();
    }

    private void bindElement() {
        binding.buttonConfirm.setOnClickListener(v -> confirmListenerClick());
    }

    private void confirmListenerClick() {
        Map<String, String> authInfo = new Authorization().getToken(
                checkLogin(binding.loginEditText.getText().toString()),
                checkPassword(binding.passwordEditText.getText().toString())
        );
        saveAuthInfo(authInfo);
    }

    private void saveAuthInfo(Map<String, String> authInfo) {
        String error = authInfo.getOrDefault(getString(R.string.error), null);
        if (error != null) {
            onException(error);
            return;
        }

        SharedPreferences preferences = requireActivity()
                .getSharedPreferences(getString(R.string.authorization_info), Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString(
                getString(R.string.token_key),
                authInfo.get(getString(R.string.token_key)));
        editor.putString(getString(R.string.user_agent),
                authInfo.get(getString(R.string.user_agent)));
        editor.apply();
    }

    private void onException(String error) {
        //logic if error from token library
        Toast.makeText(requireContext(), "Ошибка: "+error+"опробуйте в другой раз", Toast.LENGTH_SHORT).show();
    }

    private String checkLogin(String login) {
        return login;
    }

    private String checkPassword(String password) {
        return password;
    }
}

File: app/src/main/java/com/lis/player_java/ui/AuthorizationFragment.kt
package com.lis.player_java.ui

import android.app.backup.BackupAgent
import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import com.lis.player_java.vkaudiotoken.TokenReceiverOfficial
import android.widget.Toast
import com.lis.player_java.R
import android.util.Log
import android.view.View
import android.widget.EditText
import androidx.core.view.isVisible
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.Fragment
import androidx.lifecycle.lifecycleScope
import com.lis.player_java.ImageFun
import com.lis.player_java.databinding.FragmentAuthorizationBinding
import com.lis.player_java.vkaudiotoken.TokenException
import com.lis.player_java.vkaudiotoken.TokenExceptionType
import com.lis.player_java.vkaudiotoken.network.VK_OFFICIAL
import kotlinx.coroutines.launch
import kotlin.math.log

class AuthorizationFragment : DialogFragment() {
    private lateinit var binding: FragmentAuthorizationBinding
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentAuthorizationBinding.inflate(inflater, container, false)
        binding.bindElement()
        return binding.root
    }

    private fun FragmentAuthorizationBinding.bindElement() {
        getTokenButton.setOnClickListener {
            lifecycleScope.launch {
                confirmListenerClick()
            }
        }
    }

    private var captchaSid: String? = null
    private suspend fun FragmentAuthorizationBinding.confirmListenerClick() {
        val login = usernameEditText.text.toString().ifEmpty { return }
        val password = passwordEditText.text.toString().ifEmpty { return }
        val code = codeEditText.text.toString()
            .ifEmpty { if (codeEditText.visibility == View.GONE) null else return }
        val captchaKey = captchaEditText.text.toString()
            .ifEmpty { if (captchaEditText.visibility == View.GONE) null else return }
        val tokenReceiver = TokenReceiverOfficial(login, password)
        try {
            val token = tokenReceiver.getToken(code, captchaSid, captchaKey)
            saveAuthInfo(token, VK_OFFICIAL.userAgent)
            onDestroyView()
        } catch (e: TokenException) {
            when (e.code) {
                TokenExceptionType.REGISTRATION_ERROR -> {
                    errorMessage.text = e.message
                    errorMessage.visibility = View.VISIBLE
                }
                TokenExceptionType.TWO_FA_REQ -> {
                    errorMessage.text = "Было отправлено смс с кодом подтверждения"
                    errorMessage.visibility = View.VISIBLE
                    codeEditText.visibility = View.VISIBLE
                }
                TokenExceptionType.TWO_FA_ERR -> {
                    errorMessage.text = "Ошибка двухфакторной аунтификации, попробуйте снова."
                    errorMessage.visibility = View.VISIBLE
                    hideElement()
                }
                TokenExceptionType.REQUEST_ERR -> {
                    errorMessage.text = "Ошибка запроса, попробуйте снова."
                    errorMessage.visibility = View.VISIBLE
                    hideElement()
                }
                TokenExceptionType.NEED_CAPTCHA -> {
                    captchaSid = e.captchaSid
                    errorMessage.text = e.message
                    errorMessage.visibility = View.VISIBLE
                    ImageFun().setImage(captchaImage, e.captchaImg)
                    captchaLayout.visibility = View.VISIBLE
                }
                TokenExceptionType.TOKEN_NOT_RECEIVED -> {
                    errorMessage.text = "Токен небыл отправлен"
                    errorMessage.visibility = View.VISIBLE
                    hideElement()
                }
            }
        }

    }

    private fun FragmentAuthorizationBinding.hideElement() {
        codeEditText.visibility = View.GONE
        captchaLayout.visibility = View.GONE
    }

    private fun saveAuthInfo(token: String, userAgent: String) {

        val preferences = requireActivity()
            .getSharedPreferences(getString(R.string.authorization_info), Context.MODE_PRIVATE)

        if (preferences != null) {
            with(preferences.edit()) {
                putString(getString(R.string.token_key), token)
                putString(getString(R.string.user_agent), userAgent)
                apply()
            }
        }
    }
}